README.md

```diff
@@ -0,0 +1,330 @@
+# Introduction to ShrinkWrap Resolvers
+
+ShrinkWrap Resolvers is a Java API designed to simplify the process of obtaining artifacts from a repository system. It facilitates the retrieval of third-party libraries and dependent modules, allowing developers to easily incorporate them into their projects.
+
+The project primarily focuses on supporting Maven-based repository structures, enabling developers to specify coordinates that uniquely identify an artifact in the repository. It also provides basic support for Gradle-based projects.
+
+# Requirements
+
+## Build:
+
+- **JDK**: Version 8 or newer.
+- **Maven**: Version 3.6.3 or newer.
+
+## Use:
+
+- **JDK**: Version 8 or newer.
+- **Maven**: Version 3.3.9 or newer.
+- **Gradle**: Version 3 or newer.
+
+
+## Adding ShrinkWrap Resolvers to Your Project
+
+To include ShrinkWrap Resolvers in your project, you can add a dependency on the `shrinkwrap-resolver-depchain` module in your Maven `pom.xml`:
+
+```xml
+<dependencies>
+    <dependency>
+      <groupId>org.jboss.shrinkwrap.resolver</groupId>
+      <artifactId>shrinkwrap-resolver-depchain</artifactId>
+      <version>${version.shrinkwrap.resolvers}</version>
+      <scope>test</scope>
+      <type>pom</type>
+    </dependency>
+</dependencies>
+```
+
+If you are using Arquillian BOM prior to version 1.1.0.Final, ensure that you import ShrinkWrap Resolvers BOM preceding Arquillian BOM to get version 2.0.x. Importing the ShrinkWrap BOM is recommended:
+
+```xml
+<dependencyManagement>
+  <dependencies>
+    <dependency>
+      <groupId>org.jboss.shrinkwrap.resolver</groupId>
+      <artifactId>shrinkwrap-resolver-bom</artifactId>
+      <version>${version.shrinkwrap.resolvers}</version>
+      <scope>import</scope>
+      <type>pom</type>
+    </dependency>
+    <!-- Other dependencies -->
+  </dependencies>
+</dependencyManagement>
+```
+
+## Resolving Dependencies
+
+To initiate the resolution of dependencies, utilize the `org.jboss.shrinkwrap.resolver.api.maven.Maven` class. Below are typical scenarios for its usage.
+### Resolution of Artifacts by Maven Coordinates
+
+Maven coordinates follow `groupId:artifactId:[packagingType:[classifier]]:version` format. You can use these coordinates to resolve a file, where `G` is the group ID, `A` is the artifact ID, `P` is the packaging type, `C` is the classifier, and `V` is the version.
+
+#### Resolve a file using Maven coordinates:
```

Comment:
The singular file implicates that only a single file will be resolved, however, all transitive dependencies will be downloaded as well. This should be something like "Resolved file with transitive dependencies using Maven coordinates" or similar